docs/docs/Develop/Clients/typescript-client.md

```diff
@@ -0,0 +1,259 @@
+---
+title: Langflow TypeScript client
+slug: /typescript-client
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The Langflow TypeScript client integrates Langflow's capabilities into your TypeScript applications.
+
+For more information, see the [langflow-client-ts](https://github.com/datastax/langflow-client-ts/) repository.
+
+## Install the Langflow TypeScript package
+
+To install the Langflow typescript client package, use one of the following commands:
+
+<Tabs groupId="package-manager">
+<TabItem value="npm" label="npm" default>
+
+```bash
+npm install @datastax/langflow-client
+```
+
+</TabItem>
+<TabItem value="yarn" label="yarn">
+
+```bash
+yarn add @datastax/langflow-client
+```
+
+</TabItem>
+<TabItem value="pnpm" label="pnpm">
+
+```bash
+pnpm add @datastax/langflow-client
+```
+
+</TabItem>
+</Tabs>
+
+## Initialize the Langflow TypeScript client
+
+1. Import the client into your code.
+
+```typescript
+import { LangflowClient } from "@datastax/langflow-client";
+```
+
+2. Initialize a client object to interact with your server.
+The `osLangflowClient` object allows you to interact with the Langflow API.
+
+Replace `BASE_URL` and `API_KEY` with values from your deployment.
+The default Langflow base URL is `http://localhost:7860`.To create an API key, see [API keys](/configuration-api-keys).
+
+```typescript
+const baseUrl = "BASE_URL";
+const apiKey = "API_KEY";
+const osLangflowClient = new LangflowClient({ baseUrl, apiKey });
+```
+
+## Run a flow with the Langflow TypeScript client
+
+With your Langflow client initialized, submit a message to your Langflow server and receive a response.
+
+1. Create a reference to your flow with the `flowID` retrieved from Langflow.
+```typescript
+const flow = client.flow(flowId);
+```
+
+2. Run the referenced flow and pass text to it as `input`.
+```typescript
+const response = await client.flow(flowId).run(input);
+```
+
+3. This example uses the minimum values for sending a message and running your flow on a Langflow server, with no API keys.
+Replace `baseUrl` and `flowId` with values from your deployment.
+The `input` string is the message you're sending to your flow.
+<Tabs>
+<TabItem value="TypeScript" label="TypeScript" default>
+
+```typescript
+import { LangflowClient } from "@datastax/langflow-client";
+
+const baseUrl = "http://127.0.0.1:7860";
+const client = new LangflowClient({ baseUrl });
+
+async function runFlow() {
+    const flowId = "aa5a238b-02c0-4f03-bc5c-cc3a83335cdf";
+    const input = "Is anyone there?";
+
+    const response = await client.flow(flowId).run(input);
+    console.log(response);
+}
+
+runFlow().catch(console.error);
+```
+
+</TabItem>
+<TabItem value="Response" label="Response">
+
+```
+FlowResponse {
+  sessionId: 'aa5a238b-02c0-4f03-bc5c-cc3a83335cdf',
+  outputs: [ { inputs: [Object], outputs: [Array] } ]
+}
+```
+
+</TabItem>
+</Tabs>
```

Comment:
Point 3 here is just repeating the initialization and calling steps above. If the point is to show it all in one place, maybe it should be called out as a complete example, rather than the 3rd step of running a flow?